airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/up_to_date/pipeline.py

```diff
@@ -0,0 +1,322 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING
+
+import dagger
+import requests
+from connector_ops.utils import ConnectorLanguage  # type: ignore
+from pipelines.airbyte_ci.connectors.build_image.steps.python_connectors import BuildConnectorImages
+from pipelines.airbyte_ci.connectors.bump_version.pipeline import AddChangelogEntry, BumpDockerImageTagInMetadata, get_bumped_version
+from pipelines.airbyte_ci.connectors.consts import CONNECTOR_TEST_STEP_ID
+from pipelines.airbyte_ci.connectors.context import ConnectorContext, PipelineContext
+from pipelines.airbyte_ci.connectors.reports import ConnectorReport, Report
+from pipelines.consts import LOCAL_BUILD_PLATFORM
+from pipelines.dagger.actions.python.common import with_python_connector_installed
+from pipelines.helpers.execution.run_steps import STEP_TREE, StepToRun, run_steps
+from pipelines.helpers.utils import sh_dash_c
+from pipelines.models.steps import Step, StepResult, StepStatus
+
+if TYPE_CHECKING:
+    from anyio import Semaphore
+
+PACKAGE_NAME_PATTERN = r"^([a-zA-Z0-9_.\-]+)(?:\[(.*?)\])?([=~><!]=?[a-zA-Z0-9\.]+)?$"
+
+
+class CheckIsUpdateCdkCandidate(Step):
+    """Check if the connector is a candidate for migration to poetry.
```

Comment:
The plan is to work this in to the whole dependabot thing. The stuff I mentioned in the PR. Is it not possible to check in current progress?

Reply:
@bleonard we could check in current progress but what I don't get is why we should consider the `CDK` a specific dependency, different from the ones that would be updated with a single `poetry update` execution.

From your PR description if think it don't get the *else* of *do a poetry update to update everything else*.

I'd be happier merging a new command which just does `poetry update` on a connector and exports back the updated `poetry.lock` file. It would introduce a new command which has a net new feature compared to the existing `upgrade-cdk` one.

Reply:
Well I feel like we should set the CDK explicitly so doing a poetry update wouldn't fix it. It's easy to set that explicitly and clarity on CDK conpatibility is important. `*` or `^0` won't provide that. I can add the Poetry update command now if that's the request.

Reply:
> we should set the CDK explicitly 

Ahh makes sense, this is what I did not get. Let me double check with @girarda that this is what we want. 

>  I can add the Poetry update command now if that's the request.

This can be a follow up PR if the update to a specific CDK version is what we want instead of "always latest".